plugins/module_utils/common.py

```diff
@@ -181,8 +218,14 @@ def auth_set(*names):
         client = get_api_client._pool[digest]
         return client
 
+    def generate_cache_file(configuration):
+        return 'osrcp-{0}.json'.format(hashlib.sha1(get_default_cache_id(configuration)).hexdigest())
+
+    kubeclient = kubernetes.client.ApiClient(configuration)
+    cache_file = generate_cache_file(kubeclient)
+
     try:
-        client = DynamicClient(kubernetes.client.ApiClient(configuration))
+        client = DynamicClient(kubernetes.client.ApiClient(configuration), cache_file)
```

Comment:
```suggestion
        client = DynamicClient(kubeclient, cache_file)
```

Could you please add test case and changelog for this change? Thanks

Reply:
Yes, sure. I also applied this change. Thanks.